contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.2.0--3.3.0.sql

```diff
@@ -484,17 +485,19 @@ BEGIN
 
   -- Split the phrase into an array of words
   FOREACH word IN ARRAY regexp_split_to_array(phrase COLLATE "C", '\s+' COLLATE "C") COLLATE "C" LOOP
-    -- prefix term not supported
-    IF (word COLLATE C) SIMILAR TO ('%\*' COLLATE C)  THEN
-      RAISE EXCEPTION 'Prefix term not supported';
-    END IF;
     -- Append the word to the joined_text variable
     joined_text := joined_text || word || '<->';
   END LOOP;
 
   -- Remove the trailing "<->" from the joined_text
   joined_text := substring(joined_text COLLATE "C", 1, length(joined_text) - 3) COLLATE "C";
 
+  -- If the last word has the form 'lastword*', then this is a prefix term (the last word cannot be '*')
+  -- We need to rewrite the last word into 'lastword:*'
+  IF ((joined_text COLLATE "C") SIMILAR TO ('%\*' COLLATE C)) AND (NOT (joined_text COLLATE "C") SIMILAR TO ('%<->\*' COLLATE C)) AND (NOT (joined_text COLLATE "C") SIMILAR TO ('\*' COLLATE C)) THEN
```

Comment:
Did you aware similar to operator in PG is not the same with regular expression match ? refer : https://www.postgresql.org/docs/current/functions-matching.html

Reply:
Yes. I'm following the syntax in the documentation